Objects/tupleobject.c

```diff
@@ -530,6 +547,8 @@ tuple_repeat(PyObject *self, Py_ssize_t n)
     if (np == NULL)
         return NULL;
 
+    _PyTuple_RESET_HASH_CACHE(np);
```

Comment:
is there any reason not to just move this into tuple_alloc() itself instead of repeating it everywhere tuple_alloc() is called?

Reply:
I don't personally have an objection.  Most `*_alloc` functions only do allocation, not initialization, so I didn't want to confuse that.  But the cached hash is kind of a special case...